Update core dump debugging work flow (#587)
dotnet/diagnostics: documentation/debugging-coredump.md

### Getting symbols ###

Because SOS now has symbol download support (both managed PDBs and native symbols via `loadsymbols`) all that lldb requires is the host program. The host is usually `dotnet` but for self-contained applications it the .NET Core `apphost` renamed to the program/project name. These steps will handle either case and download the host lldb needs to properly diagnose a core dump. 

> First install or update the dotnet CLI symbol tool. This only needs to be down once. See this [link](https://github.com/dotnet/symstore/tree/master/src/dotnet-symbol#install) for more details. We need version 1.0.52901 or greater of dotnet-symbol installed.

Reviewer: ```suggestion
First install or update the dotnet CLI symbol tool. This only needs to be done once. See this [link](https://github.com/dotnet/symstore/tree/master/src/dotnet-symbol#install) for more details. We need version 1.0.52901 or greater of dotnet-symbol installed.
```

    ~$ dotnet tool install -g dotnet-symbol
    You can invoke the tool using the following command: dotnet-symbol
    Tool 'dotnet-symbol' (version '1.0.52901') was successfully installed.

Or update if already installed:

    ~$ dotnet tool update -g dotnet-symbol
    Tool 'dotnet-symbol' was successfully updated from version '1.0.51501' to version '1.0.52901'.

Copy the core dump to a tmp directory.

    ~$ mkdir /tmp/dump
    ~$ cp ~/coredump.32232 /tmp/dump

Download the host program for the core dump:

    ~$ dotnet-symbol --host-only /tmp/dump/coredump.32232

If the `--host-only` option is not found, update dotnet-symbol to the latest with the above step.

> If your project/program binaries are not on the machine the core dump is being loaded on, copy all them to temporary directory also.

Reviewer: ```suggestion
If your project/program binaries are not on the machine the core dump is being loaded on, copy them to a temporary directory.
```
